Source/Particles/Collision/BinaryCollision/DSMC/DSMCFunc.H

```diff
@@ -143,56 +140,64 @@ public:
             // stride (smaller set size) until we do all collisions (larger set size)
             for (index_type k = coll_idx; k < max_N; k += min_N)
             {
-                // c1k : how many times the current particle of species 1 is paired with a particle
-                // of species 2. Same for c2k.
-                const index_type c1k = (k%NI1 < max_N%NI1) ? c1 + 1: c1;
-                const index_type c2k = (k%NI2 < max_N%NI2) ? c2 + 1: c2;
+                // do not check for collision if a particle's weight was
+                // reduced to zero from a previous collision
+                if (idcpu1[ I1[i1] ]==amrex::ParticleIdCpus::Invalid ||
+                    idcpu2[ I2[i2] ]==amrex::ParticleIdCpus::Invalid ) {
+                    p_mask[pair_index] = false;
```

Comment:
@JustinRayAngus I've opted to have an `else` bracket below this instead of 
```
                    if (max_N == NI1) { i1 += min_N; }
                    if (max_N == NI2) { i2 += min_N; }
                    pair_index += min_N;
                    continue;
```
in order to avoid code duplication. Just wanted to point that out in case you wanted to make the same switch in the fusion modification PR.